File: libs/autobot-sdk-python/autobot_sdk/autobot.py
# Copyright 2025-2026 mrveiss
# SPDX-License-Identifier: Apache-2.0
# AutoBot - AI-Powered Automation Platform
# Copyright (c) 2025 mrveiss
# Author: mrveiss
"""High-level AutoBot entry point combining all resource clients."""

from __future__ import annotations

from .client import AutoBotClient
from .resources.agents import AgentsResource
from .resources.analytics import AnalyticsResource
from .resources.knowledge import KnowledgeResource
from .resources.sessions import SessionsResource


class AutoBot(AutoBotClient):
    """Top-level SDK client with resource namespaces.

    Inherits from AutoBotClient and wires up resource sub-clients::

        async with AutoBot(base_url="http://localhost:8000") as bot:
            s = await bot.sessions.list()
            stats = await bot.knowledge.stats()
    """

    def __init__(self, **kwargs) -> None:
        super().__init__(**kwargs)
        self.sessions: SessionsResource = SessionsResource(self)
        self.agents: AgentsResource = AgentsResource(self)
        self.knowledge: KnowledgeResource = KnowledgeResource(self)
        self.analytics: AnalyticsResource = AnalyticsResource(self)

File: libs/autobot-sdk-ts/src/autobot.ts
// Copyright 2025-2026 mrveiss
// SPDX-License-Identifier: Apache-2.0
// AutoBot - AI-Powered Automation Platform
// Author: mrveiss
import { AutoBotHttpClient, type ClientOptions } from "./client.js";
import { AgentsResource } from "./resources/agents.js";
import { AnalyticsResource } from "./resources/analytics.js";
import { KnowledgeResource } from "./resources/knowledge.js";
import { SessionsResource } from "./resources/sessions.js";

/**
 * Top-level SDK client with typed resource namespaces.
 *
 * ```ts
 * const bot = new AutoBot({ baseUrl: "http://localhost:8000" });
 * const result = await bot.sessions.list();
 * ```
 */
export class AutoBot extends AutoBotHttpClient {
  readonly sessions: SessionsResource;
  readonly agents: AgentsResource;
  readonly knowledge: KnowledgeResource;
  readonly analytics: AnalyticsResource;

  constructor(options: ClientOptions = {}) {
    super(options);
    this.sessions = new SessionsResource(this);
    this.agents = new AgentsResource(this);
    this.knowledge = new KnowledgeResource(this);
    this.analytics = new AnalyticsResource(this);
  }
}
